lib/parallel_tests/cucumber/scenarios.rb

```diff
@@ -27,8 +27,16 @@ def split_into_scenarios(files, tags=[])
           # Create the ScenarioLineLogger which will filter the scenario we want
           scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new(tag_expression)
 
-          # here we loop on the files map, each file will containe one or more scenario
+          # here we loop on the files map, each file will contain one or more scenario
           features ||= files.map do |path|
+            # Copying the string because it is frozen
+            path = path.dup
+            test_lines = []
+
+            # Gather up any line numbers attached to the file path
+            until path.slice(/:\d+$/).nil?
+              test_lines << path.slice!(/:\d+$/).delete(':').to_i
+            end
```

Comment:
repeating the same regex twice / need to keep the in sync ... not fun ...

```
test_lines = path.slice!(/(:\d+)+$/).to_s[0..-1].split(':').map(&:to_i)
```

Reply:
... even better ... don't dup and then:

```
path, *test_lines = path.split(/:(?=\d+)/)
test_lines.map!(&:to_i)
```

Reply:
I'll take a crack at adding a test for this behavior and then refactor it.